frontend/src/components/ErrorBoundary.jsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+
+class ErrorBoundary extends React.Component {
+  constructor(props) {

```

Comment:
Just curious, why is this ErrorBoundary alone using class component?

Reply:
Although a functional component could handle this task, I chose to implement it using a class component for simplicity and to utilize lifecycle methods like `componentDidCatch` and `getDerivedStateFromError`. While functional component solutions exist, such as those involving third-party libraries like 'react-error-boundary' or utilizing hooks like `useEffect` and event listeners, I find the class component approach simpler and more straightforward for error boundary management.